nmostesting/NMOSTesting.py

```diff
@@ -608,43 +615,59 @@ def parse_arguments():
+    msg = ""
+    return_type = "ok"
     if args.list_suites:
         for test_suite in sorted(TEST_DEFINITIONS):
-            print(test_suite)
-        sys.exit(ExitCodes.OK)
+            msg += test_suite + '\n'
     elif args.describe_suites:
         for test_suite in sorted(TEST_DEFINITIONS):
-            print(test_suite + ": " + TEST_DEFINITIONS[test_suite]["name"])
-        sys.exit(ExitCodes.OK)
-
-    if "suite" in vars(args):
-        if args.suite not in TEST_DEFINITIONS:
-            print(" * ERROR: The requested test suite '{}' does not exist".format(args.suite))
-            sys.exit(ExitCodes.ERROR)
+            msg += test_suite + ": " + TEST_DEFINITIONS[test_suite]["name"] + '\n'
+    elif "suite" in vars(args):
         if args.list_tests:
             tests = enumerate_tests(TEST_DEFINITIONS[args.suite]["class"])
             for test_name in tests:
-                print(test_name)
-            sys.exit(ExitCodes.OK)
-        if args.describe_tests:
+                msg += test_name + '\n'
+        elif args.describe_tests:
             tests = enumerate_tests(TEST_DEFINITIONS[args.suite]["class"], describe=True)
             for test_description in tests:
-                print(test_description)
+                msg += test_description + '\n'
+        elif args.suite not in TEST_DEFINITIONS:
+            msg = " * ERROR: The requested test suite '{}' does not exist".format(args.suite)
+            return_type = "error"
+        elif getattr(args, "selection", "all") not in enumerate_tests(TEST_DEFINITIONS[args.suite]["class"]):
+            msg = " * ERROR: Test with name '{}' does not exist in test definition '{}'".format(
+                args.selection, args.suite)
+            return_type = "error"
+        elif not args.host or not args.port or not args.version:
+            msg = " * ERROR: No Host(s) or Port(s) or Version(s) specified"
+            return_type = "error"
+        elif len(args.host) != len(args.port) or len(args.host) != len(args.version):
+            msg = " * ERROR: Hostnames/IPs, ports and versions must contain the same number of elements"
+            return_type = "error"
+        elif len(args.host) != len(TEST_DEFINITIONS[args.suite]["specs"]):
+            msg = " * ERROR: This test definition expects {} Hostnames/IP(s), port(s) and version(s)".format(
+                len(TEST_DEFINITIONS[args.suite]["specs"]))
+            return_type = "error"
+        elif args.output and not args.output.endswith("xml") and not args.output.endswith("json"):
+            msg = " * ERROR: Output file must end with '.xml' or '.json'"
+            return_type = "error"
+    elif access_type == "http" and "suite" not in vars(args):
+        msg = "'suite' not found in arguments"
+        return_type = "error"
+    return arg_return(access_type, return_type, msg)
+
+
+def arg_return(access_type, return_type="ok", msg=""):

```

Comment:
Could the 'return_type' argument be modified to use the existing ExitCodes enums to reduce the potential for mis-spelt strings?